src/highlighting.c

```diff
@@ -1776,3 +1776,23 @@ gboolean highlighting_is_code_style(gint lexer, gint style)
 	return !(highlighting_is_comment_style(lexer, style) ||
 		highlighting_is_string_style(lexer, style));
 }
+
+
+gchar **highlighting_get_keywords(GeanyFiletypeID filetype_id)
+{
+	GString *str = g_string_sized_new(1000);
+	gchar **keywords;
+	gchar **keyword_str;
+
+	foreach_strv(keyword_str, style_sets[filetype_id].keywords)
```

Comment:
> loops through each char in each element in the string vector?

After I manually expanded the macro, I see it uses `foreach_str` to accomplish my above loop, rather than to iterate through the chars in a C string like its document says it's meant for. I guess it was re-used with different semantics to save an additional line of normal everyday readable C code in the macro itself.